File: minigamefinal.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

let playerScore = 0;
let computerScore = 0;

function getComputerChoice() {
  const choices = ['rock', 'paper', 'scissors'];
  const randomIndex = Math.floor(Math.random() * choices.length);
  return choices[randomIndex];
}

function getUserChoice() {
  return new Promise((resolve) => {
    readline.question("Enter rock, paper, or scissors: ", (choice) => {
      if (!['rock', 'paper', 'scissors'].includes(choice)) {
        console.log("Invalid choice. Please choose rock, paper, or scissors.");
        resolve(getUserChoice()); // Recursive call to get a valid input
      } else {
        resolve(choice);
      }
    });
  });
}

function determineWinner(userChoice, computerChoice) {
  if (userChoice === computerChoice) {
    return "It's a tie!";
  } else if (
    (userChoice === "rock" && computerChoice === "scissors") ||
    (userChoice === "paper" && computerChoice === "rock") ||
    (userChoice === "scissors" && computerChoice === "paper")
  ) {
    return "You win!";
  } else {
    return "Computer wins!";
  }
}

async function playGame() {
  const userChoice = await getUserChoice();
  const computerChoice = getComputerChoice();
  console.log(`Computer chose ${computerChoice}`);
  const result = determineWinner(userChoice, computerChoice);
  console.log(result);

  if (result === "You win!") {
    playerScore += 1;
  } else if (result === "Computer wins!") {
    computerScore += 1;
  }

  readline.question("Do you want to play again? (yes/no): ", (playAgain) => {
    if (playAgain === "yes") {
      playGame();
    } else {
      console.log(`Final Scores - Player: ${playerScore}, Computer: ${computerScore}`);
      readline.close();
    }
  });
}

playGame();

File: minigamefinal.py
import random

player_score = 0
computer_score = 0

def get_computer_choice():
    return random.choice(['rock', 'paper', 'scissors'])

def get_user_choice():
    choice = input("Enter rock, paper, or scissors: ")
    if choice not in ['rock', 'paper', 'scissors']:
        print("Invalid choice. Please choose rock, paper, or scissors.")
        return get_user_choice()
    return choice

def determine_winner(user_choice, computer_choice):
    if user_choice == computer_choice:
        return "It's a tie!"
    elif (user_choice == "rock" and computer_choice == "scissors") or \
         (user_choice == "paper" and computer_choice == "rock") or \
         (user_choice == "scissors" and computer_choice == "paper"):
        return "You win!"
    else:
        return "Computer wins!"

def play_game():
    global player_score, computer_score

    user_choice = get_user_choice()
    computer_choice = get_computer_choice()
    print(f"Computer chose {computer_choice}")
    result = determine_winner(user_choice, computer_choice)
    print(result)

    if result == "You win!":
        player_score += 1
    elif result == "Computer wins!":
        computer_score += 1

    play_again = input("Do you want to play again? (yes/no): ")
    if play_again == "yes":
        play_game()
    else:
        print(f"Final Scores - Player: {player_score}, Computer: {computer_score}")

if __name__ == "__main__":
    play_game()